### ISSUE TITLE
add type to every Named

### ISSUE THREAD

**Issue by behrisch** [eclipse/sumo#11213]
This is an idea to add a type attribute (maybe reuse the XML tag) to every Named object. This came up while thinking about context subscriptions and whether we really need a separate rtree for every domain.

Pro: may help on other occasions as well where we currently need dynamic type casts or other means
Con: memory consumption (adds another int to almost every object)

Please discuss

**Comment by namdre** [comment]
How about just adding the virtual (abstract?) method `getType` to every Named object? Most objects should already be able to provide this so we don't get a memory overhead.

**Comment by behrisch** [comment]
Yes, of course. Since this basically a class property anyway there is no need to store it as an attribute. I changed the title accordingly
